[#8132] Update actions and pages which set "noindex", "nofollow" crawler directives

gbp opened the pull request:
## Relevant issue(s)

Fixes #8132

## What does this do?

Update actions and pages which set "noindex", "nofollow" crawler directives

## Why was this needed?

> Snippets of request content often appear on list pages, and create a whack-a-mole situation when unhappy users find that external search engines have indexed a list page (e.g. /body/foo?page=12) that contains a cached snippet of PII that we've removed from the request page itself.

## Implementation notes

@garethrees are you happy with changing the number of paginated pages which are indexed? I'm concerned this might impact search ranking due to newer request pages not being indexed at all.

gbp committed:
Update setting no crawl header

Extract out into a common concern included in `ApplicationController` so
this hook/helper is available to all controllers.

gbp committed:
Update all paginated pages to set no crawl headers

Pages after the first shouldn't be crawled, this will help with site
performance.

gbp committed:
Add nofollow to no crawl headers

gbp committed:
Update actions that set no crawl headers

Don't allow indexing of:
- New citation page
- New request page
- Similar requests page
- Request details page
- User profile wall
- User profile if suspended

gbp committed:
Add no crawl meta tags to HTML head

Even though we're setting the response header ensure we set a consistent
value in the meta tag.

gbp committed:
Add nofollow to request action menu links

These actions require a user to be logged in or link to actions which we
don't allow to be indexed and as such there is no reason for search
indexers to follow them.

gbp committed:
Update changelog

garethrees reviewed (commented):
Worth making the dependency on `RobotsHeaders` explicit in the concerns that use it:

```diff
  module ProminenceHeaders
    extend ActiveSupport::Concern
+    include RobotsHeaders

  module PublicTokenable
    extend ActiveSupport::Concern
+    include RobotsHeaders
```

https://api.rubyonrails.org/classes/ActiveSupport/Concern.html says you can do this (in the intro section of the documentation)

garethrees commented:
Reassigning as discussed to make a few tweaks.

gbp committed:
Add nofollow to pagination links over page 20

gbp committed:
Add nosnippet data attribute to request previews

This attribute is supported by Google and will prevent snippets from
being rendered in Google search results.

This will mean any PII cached is easier to removed from without needing
to issue search engine removal requests.

garethrees commented on app/controllers/citations_controller.rb:
```diff
@@ -5,6 +5,7 @@ class CitationsController < ApplicationController
   before_action :authenticate, except: :index
   before_action :load_resource_and_authorise, except: :index
   before_action :set_in_pro_area, except: :index
+  before_action :set_no_crawl_headers
```
Needs `except: :index`

garethrees reviewed (commented):
1 minor change but LGTM after that
